### PR TITLE
Add ALTER TABLE .. SET LOGGED/UNLOGGED support

### LINKED ISSUE #4031: ALTER TABLE ... SET LOGGED/UNLOGGED support
Currently we don't support `ALTER TABLE ... SET LOGGED/UNLOGGED` command for distributed tables.

If a table is UNLOGGED, then it will not be logged to WAL(write-ahead-log), which will be faster compared to ordinary table, but then they are not crash-safe. Currently we can use the following work-around to execute these commands on a distributed table:

```sql
BEGIN;

SET LOCAL citus.enable_ddl_propagation to OFF;

ALTER TABLE table SET UNLOGGED;

SET LOCAL citus.enable_ddl_propagation to ON;

SELECT run_command_on_placements('table', 'ALTER TABLE %s SET UNLOGGED');

COMMIT;
```

We should handle `AT_SetLogged/AT_SetUnLogged` cases in the codebase.

### PR DESCRIPTION
**Pull request by naisila**
DESCRIPTION: Add ALTER TABLE .. SET LOGGED/UNLOGGED support

TODO:

- [x] verify it is trivial to support

Fixes #4031

### CODE REVIEW

File: src/test/regress/output/multi_alter_table_statements.source [hunk 1]
```diff
@@ -1066,6 +1066,64 @@ SELECT relname FROM pg_class WHERE relname LIKE 'test_table_1%';
 (0 rows)
 
 \c - - - :master_port
+-- verify logged info is propagated to workers when distributing the table
```
Comment (SaitTalhaNisanci) [inline src/test/regress/output/multi_alter_table_statements.source:1069]: Could we verify these commands with `citus.log_remote_commands`? Have you considered that?

Comment (naisila) [reply src/test/regress/output/multi_alter_table_statements.source:1069]: I had not considered it but makes sense 👍

Comment (naisila) [reply src/test/regress/output/multi_alter_table_statements.source:1069]: Update: keeping the first approach for consistency with the old `multi_alter_table_statements` source

File: src/test/regress/output/multi_alter_table_statements.source [hunk 2]
```diff
@@ -1066,6 +1066,64 @@ SELECT relname FROM pg_class WHERE relname LIKE 'test_table_1%';
...
+CREATE TABLE logged_test(id int);
```
Comment (SaitTalhaNisanci) [inline src/test/regress/output/multi_alter_table_statements.source:1070]: I guess if you put the new tests to the end then you wouldn't have the other updates in this file. But it is a minor thing.

**Review by SaitTalhaNisanci — APPROVED** [review]

File: src/test/regress/output/multi_alter_table_statements.source (hunk 2 above)
Comment (naisila) [reply src/test/regress/output/multi_alter_table_statements.source:1070]: I see what you mean. I tried to put the tests in a place that is somewhat consistent to the rest of the file.

**Pull request merged**